### PR TITLE
Resolve issues reported by PHPStan at level 4

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring

Level 4:
- basic dead code checking
- always false `instanceof` and other type checks
- dead `else` branches
- unreachable code after return

### CODE REVIEW

**Review by kylekatarnls — DISMISSED** [review]

**Comment by AJenbo** [comment]
@kylekatarnls I had to rebase because a conflict in phpstan.neon. Could you re-approve :)

File: src/main/php/PHPMD/RuleSetFactory.php [hunk 1]
```diff
@@ -503,7 +503,10 @@ private function getPropertyValue(SimpleXMLElement $propertyNode)
     public function getIgnorePattern($fileName)
     {
         $excludes = [];
-        foreach (array_map('trim', explode(',', $fileName)) as $ruleSetFileName) {
+        $files = array_map('trim', explode(',', $fileName));
```
Comment (kylekatarnls) [inline src/main/php/PHPMD/RuleSetFactory.php:506]:
```suggestion
        $files = array_map(trim(...), explode(',', $fileName));
```

**Review by kylekatarnls — APPROVED** [review]

File: src/main/php/PHPMD/RuleSetFactory.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PHPMD/RuleSetFactory.php:506]: It's out of scope, but I'll do it in a follow up:)

Comment (AJenbo) [reply src/main/php/PHPMD/RuleSetFactory.php:506]: Here we go https://github.com/phpmd/phpmd/pull/1132

**Pull request merged**